core/arch/arm/kernel/boot.c

```diff
@@ -1423,6 +1423,33 @@ static void init_primary(unsigned long pageable_part, unsigned long nsec_entry)
 	init_sec_mon(nsec_entry);
 }
 
+#ifdef CFG_SECURE_PARTITION
+static void check_crc32_feature(void)
+{
+	static bool checked;
+	static bool available;
+
+	/* Check once (i.e. on the boot core) and store the result */
+	if (!checked) {
+		available = feat_crc32_implemented();
+		checked = true;
+		return;
+	}
+
+	/* If available on the boot core, make sure other cores match */
+	if (available) {
+		if (!feat_crc32_implemented()) {
```

Comment:
Do we really support systems where the different CPUs we execute code on are incompatible with each other?

Reply:
If so, how are the incompatibilities bounded?

Reply:
I'm not sure, it is possible in theory but I don't know if it exists in real HW. As discussed in the thread above, Jerome had concerns about it, so I added this check to catch the error if we ever run on such asymmetrical system. Should I  keep it or remove it? I'm okay with both approaches, it depends on what you and @jforissier prefer.

Reply:
Let's wait with these kinds of checks until we know they are needed if you don't mind @jforissier.

Reply:
OK but I would at least like a comment in `handle_hw_features()` to state that it is called from the primary CPU and it assumes that if the feature is present then all secondary CPUs have the feature enabled too.

Reply:
Okay, I'll update it.

Reply:
Done